keanu-project/src/main/java/io/improbable/keanu/vertices/generic/nonprobabilistic/PrintVertex.java

```diff
@@ -0,0 +1,74 @@
+package io.improbable.keanu.vertices.generic.nonprobabilistic;
+
+import com.google.common.base.Preconditions;
+import io.improbable.keanu.annotation.ExportVertexToPythonBindings;
+import io.improbable.keanu.vertices.LoadVertexParam;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.SaveVertexParam;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.generic.GenericVertex;
+import java.io.PrintStream;
+
+public class PrintVertex<T> extends GenericVertex<T> implements NonProbabilistic<T> {
+
+    public static void setPrintStream(PrintStream printStream) {
+        PrintVertex.printStream = Preconditions.checkNotNull(printStream);
+    }
+
+    private static PrintStream printStream = System.out;
+
+    private static final String PARENT = "parent";
+
+    private static final String MESSAGE = "message";
+
+    private static final String PRINT_DATA = "printData";
+
+    private final Vertex<T> parent;
+
+    private final String message;
+
+    private final boolean printData;
+
+    @ExportVertexToPythonBindings
+    public PrintVertex(@LoadVertexParam(PARENT) Vertex<T> parent,
+                       @LoadVertexParam(MESSAGE) final String message,
+                       @LoadVertexParam(PRINT_DATA) boolean printData) {
+        super(parent.getShape());
+        this.parent = parent;
+        this.message = Preconditions.checkNotNull(message);
+        this.printData = printData;
+        setParents(parent);
+    }
+
+    public PrintVertex(Vertex<T> parent) {
+        this(parent, "Calculated Vertex:\n", true);
+    }
+
+    @Override
+    public T sample(KeanuRandom random) {
+        return parent.sample();
+    }
+
+    @Override
+    public T calculate() {
+        final String dataOutput = printData ? parent.getValue().toString() + "\n" : "";
+        printStream.print(message + dataOutput);
```

Comment:
I think we could print out the vertex label for the parent as well.

Reply:
I think, as Caleb noted, there's value in keeping this vertex relatively unopinionated. The caller is free to put the vertex label into the print message, I believe the label is unlikely to change throughout the trace so it would work okay.

I'm pretty on the fence but would rather start minimal and add more things if enough people would find it useful.

Reply:
Yeah that makes sense.